### PR TITLE
QueryFile intro & QScriptCore.Map

### PR DESCRIPTION
**Pull request by rabbitonweb**
- [x] Create QueryFile template
- [x] Implement execute plan method that delegates to Planner type class
- [x] Introduce Planner template  
- [x] Partially implement QScriptCore.Map but *NOT* ready for evaluation just yet

### CODE REVIEW

**Comment by rabbitonweb** [comment]
The idea behind this PR is to push all my current work done regarding `QueryFile` to master. Non of things being implemented here are to be treated as 100% done. 

It's more to get all that stuff (27 files changed) to repo so I can keep up with changing APIs easier :)

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala [hunk 1]
```diff
@@ -17,63 +17,94 @@
...
+          qs.run.run
+        }
         case _ => ???
       }
+    } 
+
+  implicit def composedFunctor[F[_]: Functor, G[_]: Functor]:
+      Functor[(F ∘ G)#λ] =
+    new Functor[(F ∘ G)#λ] {
+      def map[A, B](fa: F[G[A]])(f: A => B) = fa ∘ (_ ∘ f)
```
Comment (sellout) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala]: You may not need this instance anymore. I think we’re using transformers in the relevant places now.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/Planner.scala [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+              maybeFunc.map(df => (sc, src.filter{
+                df >>> {
+                  case Data.Bool(b) => b
+                  case _ => false
+                }
+              })).point[Task]
+            }
+          )
+        case Take(src, from, count) =>
+          ???
```
Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/Planner.scala]: Ok, make sense. I will push that in next PR (coming right after this one :) )

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala (hunk 1 above)
Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala]: I will delete it an see what happens :)

**Comment by sellout** [comment]
Ok, small thing – the CHANGELOG should be “revision.md”, because we haven’t published MRA connectors yet.

**Comment by sellout** [comment]
Commit it with `[skip ci]` so we don’t start another build.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala (hunk 1 above)
Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala]: well, 
```
[error] /Users/rabbit/projects/quasar/sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala:88: could not find implicit value for evidence parameter of type scalaz.Functor[[A]scalaz.concurrent.Task[scalaz.WriterT[[+X]X,scala.collection.immutable.Vector[quasar.PhaseResult],A]]]
[error]         sparkStuff >>= (mrdd => mrdd.bitraverse[(Task ∘ Writer[PhaseResults, ?])#λ, FileSystemError, AFile](
[error]                                                                                                            ^
[error] /Users/rabbit/projects/quasar/sparkcore/src/main/scala/quasar/physical/sparkcore/fs/queryfile.scala:92: could not find implicit value for parameter ev: scalaz.Liskov.<~<[A,F[B]]
[error]     }.join
[error]       ^
[error] two errors foun
```

:)

**Pull request merged**